### PR TITLE
Delete item with relationships: configure defaults

### PR DESCRIPTION
**Pull request by benbosman**
This includes support for the configuration part of https://github.com/DSpace/Rest7Contract/pull/78

This replaces https://github.com/DSpace/DSpace/pull/2588

### CODE REVIEW

**Comment by paulo-graca** [comment]
@benbosman do you think the default [/dspace]/config/entities/relationships.xml should be added in this PR since some of the changes are related with that?

File: dspace-api/src/main/java/org/dspace/content/service/RelationshipTypeService.java [hunk 1]
```diff
@@ -155,4 +156,24 @@ RelationshipType create(Context context, EntityType leftEntityType, EntityType r
...
+     * This method will support the creation of a RelationshipType object with the given parameters
+     * @param context                       The relevant DSpace context
+     * @param leftEntityType                The leftEntityType EntityType object for this relationshipType
+     * @param rightEntityType               The rightEntityType EntityType object for this relationshipType
+     * @param leftwardType                  The leftwardType String object for this relationshipType
+     * @param rightwardType                 The rightwardType String object for this relationshipType
+     * @param leftCardinalityMinInteger     The leftCardinalityMinInteger Integer object for this relationshipType
+     * @param leftCardinalityMaxInteger     The leftCardinalityMaxInteger Integer object for this relationshipType
+     * @param rightCardinalityMinInteger    The rightCardinalityMinInteger Integer object for this relationshipType
+     * @param rightCardinalityMaxInteger    The rightCardinalityMaxInteger Integer object for this relationshipType
```
Comment (paulo-graca) [inline dspace-api/src/main/java/org/dspace/content/service/RelationshipTypeService.java]: missing the new parameters Boolean copyLeft, Boolean copyRight

File: dspace-api/src/main/java/org/dspace/app/util/InitializeEntities.java [hunk 2]
```diff
@@ -175,7 +189,8 @@ private String getString(String leftCardinalityMin,Element node, String minOrMax
 
     private void populateRelationshipType(Context context, String leftType, String rightType, String leftwardType,
                                           String rightwardType, String leftCardinalityMin, String leftCardinalityMax,
-                                          String rightCardinalityMin, String rightCardinalityMax)
+                                          String rightCardinalityMin, String rightCardinalityMax,
+                                          Boolean copyLeft, Boolean copyRight)
```
Comment (paulo-graca) [inline dspace-api/src/main/java/org/dspace/app/util/InitializeEntities.java]: What does it mean copyLeft and copyRight, What does it do? 
Is CopyLeft the copy record from left to right entity, or copy from right to left? 

Perhaps the naming should include "from" or "to" to indicate the direction of the copy. What do you think?

**Review by paulo-graca — COMMENTED** [review]
I just have some minor comments regarding the naming in this PR. I think it should be more clear what is copyLeft or copyRight.

**Comment by benbosman** [comment]
Thanks @paulo-graca for your feedback, we'll update the branch based on these useful improvements.

File: dspace-api/src/main/resources/org/dspace/storage/rdbms/sqlmigration/postgres/V7.0_2019.11.13__relationship_type_copy_left_right.sql [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+--
+-- http://www.dspace.org/license/
+--
+
+-----------------------------------------------------------------------------------
+-- Create columns copy_left and copy_right for RelationshipType
+-----------------------------------------------------------------------------------
+
+ALTER TABLE relationship_type ADD copy_left BOOLEAN DEFAULT FALSE NOT NULL;
+ALTER TABLE relationship_type ADD copy_right BOOLEAN DEFAULT FALSE NOT NULL;
```
Comment (tdonohue) [inline dspace-api/src/main/resources/org/dspace/storage/rdbms/sqlmigration/postgres/V7.0_2019.11.13__relationship_type_copy_left_right.sql]: I agree with @paulo-graca that the names CopyLeft and CopyRight are vague.  I just wanted to point out that we should also correct them at the database level.  These should be renamed `copy_to_left` and `copy_to_right`  (assuming I have the meaning correct).

**Review by tdonohue — COMMENTED** [review]
@benbosman : Gave this a quick review as well. Overall, the code here looks good to me.  But, I agree with @paulo-graca's note that we should rename `copyLeft` and `copyRight` to be more descriptive.  The name is also vague to me...I suspect they are meant to be `copyToLeft` and `copyToRight`.

File: dspace-api/src/main/java/org/dspace/app/util/InitializeEntities.java (hunk 2 above)
Comment (benbosman) [reply dspace-api/src/main/java/org/dspace/app/util/InitializeEntities.java]: This has been updated

File: dspace-api/src/main/resources/org/dspace/storage/rdbms/sqlmigration/postgres/V7.0_2019.11.13__relationship_type_copy_left_right.sql (hunk 3 above)
Comment (benbosman) [reply dspace-api/src/main/resources/org/dspace/storage/rdbms/sqlmigration/postgres/V7.0_2019.11.13__relationship_type_copy_left_right.sql]: This has been updated

**Comment by benbosman** [comment]
The name has been updated from copyLeft to copyToLeft (and similar for right)
The dspace-api/src/test/data/dspaceFolder/config/entities/relationship-types.xml has been updated with some sample defaults, but the default [/dspace]/config/entities/relationships.xml will be part of a separate PR as discussed in the last meeting
We also noticed a small issue in RelationshipServiceImpl.java concerning the place to maintain which has been solved and a test has been created to verify this fix

**Review by tdonohue — APPROVED** [review]
👍 The code (and tests) here look fine to me now.  Thanks for the updates!

**Comment by benbosman** [comment]
@paulo-graca if you have some time left, this could use a quick re-review
Your feedback was processed and Tim has approved it already

**Review by paulo-graca — APPROVED** [review]
Thank you @benbosman for considering my suggestions!

**Comment by benbosman** [comment]
Thanks for the 2 approvals, merging this PR

**Pull request merged**